zipkin-cassandra/src/main/scala/com/twitter/zipkin/storage/cassandra/CassandraSpanStore.scala

```diff
@@ -263,25 +263,30 @@ abstract class CassandraSpanStore(
     FutureUtil.toFuture(repository.getServiceNames).map(_.asScala.toList.sorted)
   }
 
-  override def getSpanNames(service: String): Future[Seq[String]] = {
+  override def getSpanNames(service: Option[String]): Future[Seq[String]] = {
     QueryGetSpanNamesCounter.incr()
-    FutureUtil.toFuture(repository.getSpanNames(service)).map(_.asScala.toList.sorted)
+    service match {
+      case Some(name) => FutureUtil.toFuture(repository.getSpanNames(name)).map(_.asScala.toList.sorted)
+      case None => FutureUtil.toFuture(repository.getAllSpanNames()).map(_.asScala.toList.sorted)
+    }
   }
 
   override def getTraceIdsByName(
-    serviceName: String,
+    serviceName: Option[String],
     spanName: Option[String],
     endTs: Long,
     lookback: Long,
     limit: Int
   ): Future[Seq[IndexedTraceId]] = {
     QueryGetTraceIdsByNameCounter.incr()
 
-    val traceIdsFuture = FutureUtil.toFuture(spanName match {
+    val traceIdsFuture = FutureUtil.toFuture((serviceName, spanName) match {
       // if we have a span name, look up in the service + span name index
       // if not, look up by service name only
-      case Some(x :String) => repository.getTraceIdsBySpanName(serviceName, x, endTs * 1000, lookback * 1000, limit)
-      case None => repository.getTraceIdsByServiceName(serviceName, endTs * 1000, lookback * 1000, limit)
+      case (Some(x: String), Some(y: String)) =>
+        repository.getTraceIdsBySpanName(x, y, endTs * 1000, lookback * 1000, limit)
+      case (Some(x: String), None) => repository.getTraceIdsByServiceName(x, endTs * 1000, lookback * 1000, limit)
+      case (None, y) => repository.getAllTraceIds(endTs * 1000, lookback * 1000, limit)
```

Comment:
You're not using y here, so you could use _ instead.

    case (None, _) => repository.getAllTraceIds(endTs * 1000, lookback * 1000, limit)